### ISSUE TITLE
Breaking change in 0.2.1 regarding return type from `pagy_cursor` (`ActiveRecord::Relation` -> `Array`)

### ISSUE THREAD

**Issue by walro** [Uysim/pagy-cursor#28]
Hiya, I was upgrading to 0.2.1 from 0.2.0 and noticed that the return value of `pagy_cursor` changed from a `ActiveRecord::Relation` to a plain old `Array`. I tracked down the behavior change to #19 ([diff link](https://github.com/Uysim/pagy-cursor/pull/19/files#diff-683191b7d285ea97b266d499138275502a63049912c3771310aa358685f00e86R13)) where the subscript operator causes the query to be "eagerly" evaluated.

The following test case passes on 0.2.0 but not on 0.2.1:

```diff
diff --git a/spec/pagy_cursor/cursor_spec.rb b/spec/pagy_cursor/cursor_spec.rb
index d71bb53..074462d 100644
--- a/spec/pagy_cursor/cursor_spec.rb
+++ b/spec/pagy_cursor/cursor_spec.rb
@@ -62,6 +62,12 @@ RSpec.describe Pagy::Backend do
       expect(records.last.name).to eq("user100")
       expect(pagy.has_more?).to eq(false)
     end
+
+    it 'is lazy' do
+      _, records = backend.send(:pagy_cursor, User.all)
+
+      expect(records).to be_a(ActiveRecord::Relation)
+    end
   end

   context 'with ordered records' do
```

This was a bit surprising to us (or our code-base at least) and to me it seems like it might have made the gem a bit less useful as you might not always want your records loaded just yet. I understand the change in #19 was a good one, but maybe there is a way to still support the old "chainable relation" return type?

**Comment by Uysim** [comment]
@yunanhelmy  Do you have any idea on this? Because what @walro  mention here make sense that if we don't return in ActiveRecord::Relation, it will be less useful

**Comment by yunanhelmy** [comment]
@walro Thank you for pointing out.

@Uysim That was because the return is `items[0..pagy.items-1]`, so it will become array. I will create PR to change the return not as array, but as ActiveRecord::Relation.

**Comment by yunanhelmy** [comment]
@walro @Uysim https://github.com/Uysim/pagy-cursor/pull/30 please let me know your feedback.

**Comment by Uysim** [comment]
@yunanhelmy  I think the better solution here is to revert #19

**Comment by Uysim** [comment]
@walro  and @yunanhelmy 
We have it fix at version 0.2.2
